contracts/RewardsManager.vy

```diff
@@ -1,51 +1,171 @@
-# @version 0.3.0
-# @notice A manager contract for the StakingRewards contract.
-# @author skozin
+# @version 0.3.1
+# @notice Wrapper contract for the Balancer Liquidity Gauge.
+# @author bulbozaur <info@lido.fi>
 # @license MIT
+
 from vyper.interfaces import ERC20
 
 
-interface StakingRewards:
-    def periodFinish() -> uint256: view
-    def notifyRewardAmount(reward: uint256, rewardHolder: address): nonpayable
+struct BalancerReward:
+    token: address
+    distributor: address
+    period_finish: uint256
+    rate: uint256
+    last_update: uint256
+    integral: uint256
+
+
+interface BalancerLiquidityGauge:
+    def reward_data(addr: address) -> BalancerReward: view
+    def deposit_reward_token(_reward_token: address, _amount: uint256): nonpayable
+    def set_reward_distributor(_reward_token: address, _distributor: address): nonpayable
+
+
+event OwnershipTransferred:
+    previousOwner: indexed(address)
+    newOwner: indexed(address)
+
+
+event RewardsContractUpdated:
+    newRewardsContract: indexed(address)
+
+
+event RewardsContractTransferred:
+    newDistributor: indexed(address)
+
+
+event WeeklyRewardsAmountUpdated:
+    newWeeklyRewardsAmount: uint256
+
+
+event NewRewardsPeriodStarted:
+    amount: uint256
+
+
+event ERC20Recovered:
+    token: indexed(address)
+    amount: uint256
+    recipient: indexed(address)
 
 
 owner: public(address)
 rewards_contract: public(address)
-ldo_token: constant(address) = 0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32
+min_rewards_amount: public(uint256)
+weekly_amount: public(uint256)
+rewards_iteration: public(uint256)
+LDO_TOKEN: constant(address) = 0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32
+SECONDS_PER_WEEK: constant(uint256) = 7 * 24 * 60 * 60
+WEEKS_PER_PERIOD: constant(uint256) = 4
 
 
 @external
-def __init__():
-    self.owner = msg.sender
+def __init__(
```

Comment:
Who is the initial reward distributor? Do we need to set him up during the construction stage?